recipes/b2/portable/conanfile.py

```diff
@@ -45,24 +47,31 @@ class B2Conan(ConanFile):
             'acc', 'borland', 'clang', 'como', 'gcc-nocygwin', 'gcc',
             'intel-darwin', 'intel-linux', 'intel-win32', 'kcc', 'kylix',
             'mingw', 'mipspro', 'pathscale', 'pgi', 'qcc', 'sun', 'sunpro',
-            'tru64cxx', 'vacpp', 'vc12', 'vc14', 'vc141', 'vc142', 'vc143']
+            'tru64cxx', 'vacpp', 'vc12', 'vc14', 'vc141', 'vc142', 'vc143',
+        ]
     }
     default_options = {
         'use_cxx_env': False,
         'toolset': 'auto'
     }
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        del self.info.options.use_cxx_env
+        del self.info.options.toolset
+
     def validate(self):
+        if hasattr(self, "settings_build") and cross_building(self):
+            raise ConanInvalidConfiguration(f"{self.ref} recipe doesn't support cross-build yet")
```

Comment:
I'd love to have "contributions welcome" on message like this to make it move inviting pretty please 🙏